### ISSUE TITLE
Handle NULL parameters like MySQL

### ISSUE THREAD

**Issue by tirsen** [youtube/vitess#3132]
When you pass in a NULL parameter to a prepared statement query MySQL handles that just fine. Vitess on the other hand gives the following error:
```
Caused by: java.sql.SQLNonTransientException: Vitess RPC error: legacy_code: UNKNOWN_ERROR_LEGACY
message: "vtgate: http://localhost:15001/: paramsSelectEqual: hash.Map: could not parse value: "
code: UNKNOWN
```

**Comment by tirsen** [comment]
(Technically you should be using an `IS NULL` clause in this case but it wouldn't be the first time that MySQL diverges from the SQL standard.)

**Comment by sougou** [comment]
I think the vitess behavior for the use case you showed is correct. It's not able to map a NULL value to a keyspace id. Vitess allows nulls as bind vars, but they should be used with care. For example, `where id = NULL` is always `FALSE` :sob:. I actually looked at doctoring equality expressions to `is null`, but realized the risk because values are used in so many other constructs, like `IN` clauses, etc. So, there's no way to uniformly extend the null handling semantics for all constructs.

**Comment by sougou** [comment]
This is now addressed via #3182
